squarelet/payments/views.py

```diff
@@ -19,10 +19,33 @@
 logger = logging.getLogger(__name__)
 
 
+def protect_private_plan(plan, user):
+    """Raise 404 if user should not access this private plan"""
+    if plan.private_organizations.exists():
+        # If user is not authenticated, raise 404
+        if not user.is_authenticated:
+            raise Http404("Plan not found")
+
+        # If user is not admin of any organization for this plan, raise 404
+        user_admin_orgs = user.organizations.filter(memberships__admin=True)
+        if not plan.private_organizations.filter(pk__in=user_admin_orgs).exists():
+            raise Http404("Plan not found")
+
+
 class PlanDetailView(DetailView):
     model = Plan
     template_name = "payments/plan.html"
 
+    def get_object(self, queryset=None):
+        """Override to check private plan access"""
+        if queryset is None:
+            queryset = self.get_queryset()
+
+        plan = super().get_object(queryset)
+
+        protect_private_plan(plan, self.request.user)
+        return plan
+
     def get_context_data(self, **kwargs):
         # pylint: disable=too-many-locals
         context = super().get_context_data(**kwargs)
```

Comment:
Do we need to limit `admin_orgs` to only orgs the user is an admin in which are also in `private_organizations` if `private_organizations` is set?  Otherwise if we whitelist a plan for one organization, an admin for that organization could subscribe another organization they are an admin of for that plan.